server/plugins/LFUGens.cpp

```diff
@@ -1195,20 +1195,24 @@ void SyncSaw_Ctor(SyncSaw* unit) {
 
 struct K2A : SIMD_Unit {
     ControlRateInput<0> mLevel;
+    ControlRateInput<1> mRamp;
 
     K2A(void) {
         mLevel.init(this);
+        mRamp.init(this);
         if (inRate(0) == calc_ScalarRate)
             set_unrolled_calc_function<K2A, &K2A::next_i<unrolled_64>, &K2A::next_i<unrolled>, &K2A::next_i<scalar>>();
         else
             set_unrolled_calc_function<K2A, &K2A::next_k<unrolled_64>, &K2A::next_k<unrolled>, &K2A::next_k<scalar>>();
     }
 
     template <int type> void next_k(int inNumSamples) {
-        if (mLevel.changed(this))
+        if (mRamp && mLevel.changed(this))
             slope_vec<type>(out(0), mLevel.slope(this), inNumSamples);
```

Comment:
Achieving the float-valued ramp behavior referenced in the OP might be something like (best guess without testing it!):
```
int nRampSamps = (int) mRamp * inNumSamples;
slope_vec<type>(out(0), mLevel.slope(this), nRampSamps);
slope_vec<type>(out(nRampSamps), 0.f, inNumSamples - nRampSamps);
```
bounds of `nRampSamps` would of course need to be checked